templates/prescription/general_list.html

```diff
@@ -84,31 +84,25 @@
 }
 
 function deleteDrug(d) {
-    let isPromise = top.jsFetchGlobals('e_rx');
-    isPromise.then(
-        msg => {
-            top.restoreSession();
-            alert('msg.e_rx.deleteWarn');
-            if (choice == true) {
-                $.ajax({
-                    url: "./library/deletedrug.php",
-                    type: 'POST',
-                    data: {
-                        drugId: d,
-                        csrf_token_form: {$CSRF_TOKEN_FORM|js_escape}
-                    },
-                    success: function(data) {
-                        console.log(data);
-                        location.reload();
-                    },
-                    error: function(error) {
-                        console.log(error)
-                    },
-                });
-            }
-        }).catch(error => {
-        console.log(error.message)
+    let msg = {/literal}{xlj t='Do you really want to delete?'}{literal};
+    let choice = confirm(msg + d);
+    if (choice == true) {
+        $.ajax({
+                url: "./library/deleteDrug.php",
+                type: 'POST',
+                data: {
+                    drugId: d,
+                    csrf_token_form: {$CSRF_TOKEN_FORM|js_url}
+            },
+            success: function(data) {
+            console.log(data);
+            location.reload();
+        },
+        error: function(error) {
+            console.log(error)
+        },
     });
```

Comment:
@juggernautsei 
Normally I wouldn't be to concerned about session on a straight ajax call however, since you are reloading the form afterwards then, you'd want to ensure a good session before call.
You could add a callback in ajax:
 beforeSend: function(){
     top.restoreSession();
   },
or just add top.restoreSession(); somewhere before ajax.

I did leave a comment couple days ago but you must have missed. Sorry

Reply:
just noted this comment after I also placed need for `top.restoreSession();` :)